test/e2e/plugin_sync/plugin_sync_lifecycle_test.go

```diff
@@ -0,0 +1,370 @@
+			Expect(ok).To(BeTrue(), "plugin group is not exist in the map")
+			Expect(len(pluginsToGenerateCRs) > numberOfPluginsToInstall).To(BeTrue(), "we don't have enough plugins in local test central repo")
+			pluginsInfoForCRsApplied, pluginCRFilePaths, err = framework.CreateTemporaryCRsForPluginsInGivenPluginGroup(pluginsToGenerateCRs[:numberOfPluginsToInstall])
+			Expect(err).To(BeNil(), "should not get any error while generating CR files")
+			ApplyConfigOnKindCluster(tf, clusterInfo, pluginCRFilePaths)
+		})
+
+		// Test case: c. create context and make sure context has created
+		It("create context with kubeconfig and context", func() {
+			By("create context with kubeconfig and context")
+			contextName = "sync-e2e-" + framework.RandomString(4)
+			err := tf.ContextCmd.CreateConextWithKubeconfig(contextName, clusterInfo.KubeConfigPath, clusterInfo.ClusterKubeContext)
+			Expect(err).To(BeNil(), "context should create without any error")
+			active, err := tf.ContextCmd.GetActiveContext(string(types.TargetK8s))
+			Expect(err).To(BeNil(), "there should be a active context")
+			Expect(active).To(Equal(contextName), "the active context should be recently added context")
+		})
+		// Test case: d. list plugins and validate plugins info, make sure all plugins installed for which CR's has applied to KIND cluster
+		It("list plugins and validate plugins being installed after context being created", func() {
+			pluginsList, err = tf.PluginCmd.ListPluginsForGivenContext(contextName, true)
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(len(pluginsList)).Should(Equal(len(pluginsInfoForCRsApplied)), "number of plugins should be same as number of plugins CRs applied")
+			Expect(framework.CheckAllPluginsExists(pluginsList, pluginsInfoForCRsApplied)).Should(BeTrue(), " plugins being installed and plugins info for which CRs applied should be same")
+		})
+
+		// Test case: e. uninstall one of the installed plugin, make sure plugin is uninstalled,
+		//				run plugin sync, make sure the uninstalled plugin has installed again.
+		It("Uninstall one of the installed plugin", func() {
+			pluginToUninstall := pluginsInfoForCRsApplied[0]
+			err := tf.PluginCmd.UninstallPlugin(pluginToUninstall.Name, pluginToUninstall.Target)
+			Expect(err).To(BeNil(), "should not get any error for plugin uninstall")
+
+			latestPluginsInstalledList := pluginsInfoForCRsApplied[1:]
+			pluginsList, err = tf.PluginCmd.ListPluginsForGivenContext(contextName, true)
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(len(pluginsList)).Should(Equal(len(latestPluginsInstalledList)), "number of plugins should be same as number of plugins CRs applied")
+			Expect(framework.CheckAllPluginsExists(pluginsList, latestPluginsInstalledList)).Should(BeTrue(), " plugins being installed and plugins info for which CRs applied should be same")
+
+			_, err = tf.PluginCmd.Sync()
```

Comment:
Before the sync and after the uninstall, Can we also check that the `tanzu plugin list` shows that `pluginToUninstall` is `not installed`?